server/lib/github-client.js

```diff
@@ -1,12 +1,20 @@
-const host = 'http://cyf-github-api.herokuapp.com';
 const fetch = require('isomorphic-fetch');
+const gitApi = 'https://api.github.com';
 
-function fetchFromGithub (path) {
-	// write something that will fetch json information from github
+fetchFromGithub = path => {
+  return (
+    fetch(`${gitApi}${path}`)
+    .then(apiRes => apiRes.json())
+    .then(json => json)
+    .catch(err => console.error(err))
+    );
 }
 
-module.exports.getUserProfile = function (username) {
-	return fetchFromGithub(`/users/${username}`);
-};
+module.exports.getUserProfile = username => fetchFromGithub(`/users/${username}`);
 
-// add a method to retrieve user pull requests here
+module.exports.getUserPullRequests = username => {
+  return (
+    fetchFromGithub(`/users/${username}/events/public`)
+    .then(data => data.filter(value => value.type === 'PullRequestEvent'))
```

Comment:
filterByEventTypeCurr = (data) => (eventType) => data.filter(value => value.type === eventType);

module.exports = {
  getUserProfile: username => fetchFromGithub(`/users/${username}`),
  getUserPullRequests: username => {
  return (
    fetchFromGithub(`/users/${username}/events/public`)
      .then(data => {
        const filterByEventType = filterByEventTypeCurr(data);
        return filterByEventType('PullRequestEvent')
      })
    );
  }
}

Hi @leggsimon 
could you please have a look on currying function above 
is it working like this?